servers/zms/src/main/java/com/yahoo/athenz/zms/store/jdbc/JDBCConnection.java

```diff
@@ -3751,6 +3765,65 @@ public DomainRoleMembers listDomainRoleMembers(String domainName) {
         return listDomainRoleMembersWithQuery(domainName, SQL_GET_DOMAIN_ROLE_MEMBERS, "listDomainRoleMembers");
     }
 
+    @Override
+    public DomainRoleMember getAllRoles(String principal, String domainName) {
+        final String caller = "getAllRoles";
+
+        int principalId = getPrincipalId(principal);
+        if (principalId == 0) {
+            throw notFoundError(caller, ZMSConsts.OBJECT_PRINCIPAL, principal);
+        }
+
+        DomainRoleMember roleMember = new DomainRoleMember();
+        roleMember.setMemberRoles(new ArrayList<>());
+        roleMember.setMemberName(principal);
+        if (StringUtil.isEmpty(domainName)) {
+            try (PreparedStatement ps = con.prepareStatement(SQL_GET_ALL_ROLES)) {
+                ps.setInt(1, principalId);
+                return getRolesForPrincipal(caller, roleMember, ps);
+            } catch (SQLException ex) {
+                throw sqlError(ex, caller);
+            }
+        } else {
+            int domainId = getDomainId(domainName);
```

Comment:
we should. handle the case where we have an invalid domain:

        if (domainId == 0) {
            throw notFoundError(caller, ZMSConsts.OBJECT_DOMAIN, domainName);
        }

and we should have a unit test case for this as well.

Reply:
Done